lib/src/working_copy.rs

```diff
@@ -786,6 +675,131 @@ impl TreeState {
         Ok(has_changes || fsmonitor_clock_needs_save)
     }
 
+    fn visit_file(
```

Comment:
At this point (maybe it changes later in the stack?), a `work_item` corresponds to one *directory*, so call it `visit_directory()` instead?